tests/MongoDB.Driver.Examples/ClientEncryptionExamples.cs

```diff
@@ -0,0 +1,138 @@
+
+            var mongoClientSettings = new MongoClientSettings
+            {
+                AutoEncryptionOptions = autoEncryptionOptions
+            };
+            var client = new MongoClient(mongoClientSettings);
+            var database = client.GetDatabase("test");
+            database.DropCollection("coll");
+            var collection = database.GetCollection<BsonDocument>("coll");
+
+            collection.InsertOne(new BsonDocument("encryptedField", "123456789"));
+
+            var result = collection.Find(FilterDefinition<BsonDocument>.Empty).First();
+            _output.WriteLine(result.ToJson());
+        }
+
+        [Fact]
+        public void ClientSideEncryptionAutoEncryptionSettingsTour()
+        {
+            RequireServer.Check().Supports(Feature.ClientSideEncryption);
+
+            var localMasterKey = Convert.FromBase64String(LocalMasterKey);
+
+            var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
+            var localKey = new Dictionary<string, object>
+            {
+                { "key", localMasterKey }
+            };
+            kmsProviders.Add("local", localKey);
+
+            var keyVaultNamespace = CollectionNamespace.FromFullName("admin.datakeys");
+            var keyVaultMongoClient = new MongoClient();
+            var clientEncryptionSettings = new ClientEncryptionOptions(
+                keyVaultMongoClient,
+                keyVaultNamespace,
+                kmsProviders);
+
+            var clientEncryption = new ClientEncryption(clientEncryptionSettings);
+            var dataKeyId = clientEncryption.CreateDataKey("local", new DataKeyOptions(), CancellationToken.None);
+            var base64DataKeyId = Convert.ToBase64String(GuidConverter.ToBytes(dataKeyId, GuidRepresentation.Standard));
+            clientEncryption.Dispose();
+
+            var collectionNamespace = CollectionNamespace.FromFullName("test.coll");
+
+            var schemaMap = $@"{{
+                properties: {{
+                    encryptedField: {{
+                        encrypt: {{
+                            keyId: [{{
+                                '$binary' : {{
+                                    'base64' : '{base64DataKeyId}',
+                                    'subType' : '04'
+                                }}
+                            }}],
+                        bsonType: 'string',
+                        algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic'
+                        }}
+                    }}
+                }},
+                'bsonType': 'object'
+            }}";
+            var autoEncryptionSettings = new AutoEncryptionOptions(
+                keyVaultNamespace,
+                kmsProviders,
+                schemaMap: new Dictionary<string, BsonDocument>()
+                {
+                    { collectionNamespace.ToString(), BsonDocument.Parse(schemaMap) }
```

Comment:
Have you executed this code?

I'm concerned that `BsonDocument.Parse` might mess up the `keyId` value due to `GuidRepresentation` issues.

Reply:
>Have you executed this code?

yes, this code inserts the encrypted data(line `132`) in the `test.coll` and reads the decrypted data(line `134`).

NOTE: we use `BsonDocument.Parse` in the encryption prose test: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs#L344

Taking into account my above comment, do you think `BsonDocument.Parse` should be additionally checked?

Reply:
If you have verified that we have no `GuidRepresentation` issues in this code then it's fine.